tools/distpartitioning/data_shuffle.py

```diff
@@ -624,16 +737,17 @@ def gen_dist_partitions(rank, world_size, params):
     memory_snapshot("DataShuffleComplete: ", rank)
 
     #sort node_data by ntype
-    idx = node_data[constants.NTYPE_ID].argsort()
-    for k, v in node_data.items():
-        node_data[k] = v[idx]
-    idx = None
+    for local_part_id in range(params.num_parts//world_size):
+        idx = node_data[constants.NTYPE_ID+"/"+str(local_part_id)].argsort()
+        for k, v in node_data.items():
+            if k.endswith(str(local_part_id)):
```

Comment:
if we have more than 10 partitions, how to distinguish `node_type/feat/1` and `node_type/feat/11`? we'd better use `k.endswith('/'+str())`.

Reply:
Replaced the code so that now we compare local_part_id from the dictionary key with the variable from the for loop. Similarly replaced all the code instances where endswith was used and replaced these instances with the above logic. Now added a test case where world_size = 1, num_parts = 11 and num_chunks = 11.